domain-data/src/main/java/org/triplea/domain/data/SystemIdLoader.java

```diff
@@ -0,0 +1,60 @@
+package org.triplea.domain.data;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.prefs.BackingStoreException;
+import java.util.prefs.Preferences;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.Setter;
+import lombok.experimental.UtilityClass;
+import lombok.extern.java.Log;
+
+/** Loads a SystemId from persistence. */
+@UtilityClass
+@Log
+public class SystemIdLoader {
+  private static final String SYSTEM_KEY = "system-id-key";
+
+  @Setter(value = AccessLevel.PACKAGE, onMethod_ = @VisibleForTesting)
+  private static PreferencesPersistence preferencesPersistence =
```

Comment:
The code in this class is largely just moved from [SystemIdHeader](https://github.com/triplea-game/triplea/pull/5622/files#diff-81cfc0624ee3ee51ecd9cea802dd76c1), and instead of creating a Map of system id key to system id, this class just focuses on returning (and/or generating) a system id key.